src/PhpDoc/PhpDocNodeResolver.php

```diff
@@ -452,6 +452,17 @@ public function resolveIsImpure(PhpDocNode $phpDocNode): bool
 		return false;
 	}
 
+	public function resolveIsReadOnly(PhpDocNode $phpDocNode): bool
+	{
+		foreach ($phpDocNode->getTags() as $phpDocTagNode) {
+			if (in_array($phpDocTagNode->name, ['@readonly', '@psalm-readonly', '@phpstan-readonly'], true)) {
```

Comment:
Would be nicer to match `resolveHasConsistentConstructor` below here - `getTagsByName` + `count`.